modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationEndpoints/CreateAnAPIApplicationEndpoint.testcase

```diff
@@ -94,19 +111,276 @@ definition {
+			AssertElementPresent(locator1 = "APIBuilder#SLASH");
+		}
+	}
+
+	@priority = 4
+	test CanSeeScopeOfScopeFields {
+		property portal.acceptance = "true";
+
+		task ("And Given I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("And Given I click 'Add API Endpoint' button") {
+			Click(locator1 = "Button#PLUS");
+		}
+
+		task ("When I click 'Create' button in 'New API Endpoint'") {
+			Click(
+				key_value = "Select Scope",
+				locator1 = "APIBuilder#CREATE_BUTTON");
+		}
+
+		task ("Then Then I click Scope option") {
+			Click(
+				key_value = "Select Scope",
+				locator1 = "APIBuilder#DROPDOWN_TOGGLE");
+		}
+
+		task ("And Then I can see site and company scope in the Scope option") {
+			AssertElementPresent(
+				key_optionName = "Site",
+				locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+
+			AssertElementPresent(
+				key_optionName = "Company",
+				locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+		}
+	}
+
+	@priority = 4
+	test CanSeeValidationMessageForRequiredFields {
+		property portal.acceptance = "true";
+
+		task ("And Given I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("And Given I click 'Add API Endpoint' button") {
+			Click(locator1 = "Button#PLUS");
+		}
+
+		task ("When I click 'Create' button in 'New API Endpoint'") {
+			Click(locator1 = "APIBuilder#CREATE_BUTTON");
+		}
+
+		task ("Then I can see error message about required Scope and Path") {
+			AssertElementPresent(
+				key_text = "Please select a scope.",
+				locator1 = "APIBuilder#WARNING_MESSAGE_IN_DIALOG");
+
+			AssertElementPresent(
+				key_text = "Please enter a path.",
+				locator1 = "APIBuilder#WARNING_MESSAGE_IN_DIALOG");
+		}
+
+		task ("And Then no endpoint is being created") {
+			Click(locator1 = "Modal#CLOSE_BUTTON");
+
+			AssertElementPresent(
+				key_entryName = "No API Endpoint Found",
+				locator1 = "ObjectAdmin#VIEW_ENTRY_NAME_ON_PAGE");
+		}
+	}
+
+	@priority = 4
+	test EndpointFieldsAreVisibleInCreationModal {
+		property portal.acceptance = "true";
+
+		task ("When I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("Then 'Add API Enpoint' and plus buttons are present") {
+			AssertElementPresent(locator1 = "Button#PLUS");
+		}
+
+		task ("And Then I can see 'No API Endpoints found' message") {
+			AssertElementPresent(
+				key_entryName = "No API Endpoint Found",
+				locator1 = "ObjectAdmin#VIEW_ENTRY_NAME_ON_PAGE");
+		}
+	}
+
+	@priority = 4
+	test EndpointIsNotCreatedAfterClickingCancel {
+		property portal.acceptance = "true";
+
+		task ("And Given I go to 'Endpoints' tab in 'Edit API Application'") {
```

Comment:
Please help move the `task ("And Given I go to 'Endpoints' tab in 'Edit API Application'")` into setUp() since all cases are using it, thanks.